app/src/main/java/com/eveningoutpost/dexdrip/watch/miband/MiBandService.java

```diff
@@ -91,69 +102,45 @@
+    private static final long BG_UPDATE_NO_DATA_INTERVAL = 30 * Constants.MINUTE_IN_MS; //minutes
     private static final long CONNECTION_TIMEOUT = 5 * Constants.MINUTE_IN_MS; //minutes
     private static final long RESTORE_NIGHT_MODE_DELAY = (Constants.SECOND_IN_MS * 7);
     private static final int QUEUE_EXPIRED_TIME = 30; //second
     private static final int QUEUE_DELAY = 0; //ms
     private static final int CALL_ALERT_DELAY = (int) (Constants.SECOND_IN_MS * 10);
+    private static final int MESSAGE_DELAY = (int) (Constants.SECOND_IN_MS * 5);
+    static BatteryInfo batteryInfo = new BatteryInfo();
+    static private long bgWakeupTime;
 
+    static {
+        huntCharacterstics.add(Const.UUID_CHAR_HEART_RATE_MEASUREMENT);
+    }
+
+    private final PoorMansConcurrentLinkedDeque<QueueMessage> messageQueue = new PoorMansConcurrentLinkedDeque<>();
     private Subscription authSubscription;
     private Subscription notifSubscriptionDeviceEvent;
     private Subscription notifSubscriptionHeartRateMeasurement;
+    private Subscription notifSubscriptionStepsMeasurement;
     private AuthMessages authorisation;
     private Boolean isNeedToCheckRevision = true;
     private Boolean isNeedToAuthenticate = true;
-
     private boolean isWaitingCallResponce = false;
-    private Boolean isWaitingSnoozeResponce = false;
     private Boolean isNeedToRestoreNightMode = false;
     private boolean isNightMode = false;
-
-    static BatteryInfo batteryInfo = new BatteryInfo();
     private FirmwareOperations firmware;
     private Subscription watchfaceSubscription;
     private MediaPlayer player;
-
     private PendingIntent bgServiceIntent;
-    static private long bgWakeupTime;
-    private MiBand.MiBandType prevDeviceType = UNKNOWN;
-    private final PoorMansConcurrentLinkedDeque<QueueMessage> messageQueue = new PoorMansConcurrentLinkedDeque<>();
+    private MiBandType prevDeviceType = MiBandType.UNKNOWN;
     private QueueMessage queueItem;
-    private int defaultSnoozle;
-
-    public class QueueMessage {
-        @Getter
-        private String functionName;
-        @Getter
-        private String message_type = "";
-        @Getter
-        private String message = "";
-        @Getter
-        private String title = "";
-        @Getter
-        private int defaultSnoozle = 0;
-
-        public QueueMessage(String functionName) {
-            this.functionName = functionName;
-        }
-
-        public QueueMessage(String functionName, String message_type, String message, String title, int defaultSnoozle) {
-            this(functionName);
-            this.message_type = message_type;
-            this.message = message;
-            this.title = title;
-            this.defaultSnoozle = defaultSnoozle;
-        }
-    }
-
-    public enum MIBAND_INTEND_STATES {
-        UPDATE_PREF_SCREEN,
-        UPDATE_PREF_DATA
-    }
+    private BroadcastReceiver statusReceiver;
+    private String statusIOB = "";
```

Comment:
Done

Reply:
Ah, the default initialized value for String is null, which is not suitable in this case because the value can be not initialized in some cases and this will return an error